onnxruntime/core/language_interop_ops/torch/torch_proxy.cc

```diff
@@ -0,0 +1,252 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <Python.h>
+#include "core/framework/tensorprotoutils.h"
+#include "core/language_interop_ops/torch/custom_function_register.h"
+#include "core/language_interop_ops/torch/torch_proxy.h"
+#include "core/platform/env.h"
+#include "core/util/dlpack_convertor.h"
+
+namespace onnxruntime {
+namespace language_interop_ops {
+namespace torch {
+
+// Holder of GIL state.
+// It automatically acquire the state upon creation and
+// release the acquired state after being destroyed.
+class GilGuard {
+ public:
+  GilGuard() : state_(PyGILState_Ensure()){};
+  ~GilGuard() { PyGILState_Release(state_); };
+
+ private:
+  PyGILState_STATE state_;
+};
+
+void DlpackCapsuleDestructor(PyObject* data) {
+  DLManagedTensor* dlmanged_tensor = (DLManagedTensor*)PyCapsule_GetPointer(data, "dltensor");
+  if (dlmanged_tensor) {
+    // the dlmanged_tensor has not been consumed, call deleter ourselves.
+    dlmanged_tensor->deleter(const_cast<DLManagedTensor*>(dlmanged_tensor));
+  } else {
+    // the dlmanged_tensor has been consumed,
+    // PyCapsule_GetPointer has set an error indicator.
+    PyErr_Clear();
+  }
+}
+
+bool ExtractPointerOutput(PyObject* pyObj, std::vector<void*>& outputs) {
+  void* prt = PyLong_AsVoidPtr(pyObj);
+  outputs.push_back(prt);
+  return true;
+}
```

Comment:
We have some plan to pass the object back, instead of passing the address only. Wei-sheng has some local changes, have some issues to fix last time synced iwht him. @wschin , comments?

Reply:
Fixed in latest comments.